### PR TITLE
MRG: Restructure fNIRS IO tutorial

### PR DESCRIPTION
**Pull request by rob-luke**
#### Reference issue
None

#### What does this implement/fix?
Reordering and expanding the information in the fNIRS data type tutorial. As its had lots of growth lately but not much structure.

### CODE REVIEW

File: tutorials/io/30_reading_fnirs_data.py [hunk 1]
```diff
@@ -6,39 +6,78 @@
...
+             Manual modification of channel names is not recommended.
+
+
+.. _import-snirf:
+
+SNIRF (.snirf)
+================================
+
+The Shared Near Infrared Spectroscopy Format
+(`SNIRF <https://github.com/fNIRS/snirf/blob/master/snirf_specification.md>`_)
```
Comment (larsoner) [inline tutorials/io/30_reading_fnirs_data.py:37]: Making this anonymous will make CircleCI happy
```suggestion
(`SNIRF <https://github.com/fNIRS/snirf/blob/master/snirf_specification.md>`__)
```

File: tutorials/io/30_reading_fnirs_data.py [hunk 2]
```diff
@@ -6,39 +6,78 @@
...
+the data as the appropriate type.
+
+.. warning:: MNE expects a specific formatting of channel names.
+             Manual modification of channel names is not recommended.
+
+
+.. _import-snirf:
+
+SNIRF (.snirf)
+================================
```
Comment (larsoner) [inline tutorials/io/30_reading_fnirs_data.py:34]: Sphinx complains about header levels, so I would add a second-level header `***` before this to make sure this `====` lands as a third-level header (also the existing underline is too long):
```suggestion
*****************
Standardized data
*****************

SNIRF (.snirf)
==============
```

**Comment by rob-luke** [comment]
@agramfort @drammock @larsoner please merge if happy.

**Review by agramfort — APPROVED** [review]

**Comment by larsoner** [comment]
Thanks @rob-luke !

File: tutorials/io/30_reading_fnirs_data.py [hunk 3]
```diff
@@ -6,39 +6,85 @@
...
+channel location as the midpoint between source and detector.
+MNE stores the location of the channels, sources, and
 detectors.
+There are a variety of fNIRS data types which can be represented in MNE.
+For continuous wave fNIRS data this includes amplitude, optical density,
+oxyhaemoglobin, and deoxyhemoglobin.
+And for frequency domain fNIRS this additionally includes
+AC amplitude and phase.
+Different vendors save the data as different data types, and MNE will load
+the data as the appropriate type.
```
Comment (drammock) [inline tutorials/io/30_reading_fnirs_data.py:25]:
```suggestion
fNIRS devices consist of two kinds of optodes: light sources (AKA "emitters" or
"transmitters") and light detectors (AKA "receivers"). Channels are defined as
source-detector pairs, and channel locations are defined as the midpoint
between source and detector.

MNE-Python provides functions for reading fNIRS data and optode locations from
several file formats. Regardless of the device manufacturer or file format,
MNE-Python's fNIRS functions will internally store the measurement data and its
metadata in the same way (e.g., data values are always converted into SI
units). Supported measurement types include amplitude, optical density,
oxyhaemoglobin concentration, and deoxyhemoglobin concentration (for continuous
wave fNIRS), and AC amplitude and phase (for frequency domain fNIRS).
```

**Review by drammock — COMMENTED** [review]
dang... was in the middle of a review.  well, this is as far as I got.

**Pull request merged**